hazelcast/src/main/java/com/hazelcast/query/SqlPredicate.java

```diff
@@ -56,6 +57,8 @@
 public class SqlPredicate
         implements IndexAwarePredicate, VisitablePredicate, IdentifiedDataSerializable {
 
+    private static final boolean SKIP_INDEX_ENABLED = !Boolean.getBoolean("hazelcast.query.disableSkipIndex");
```

Comment:
Influenced by [this document](https://hazelcast.atlassian.net/wiki/spaces/EN/pages/253953578/Hazelcast+Configuration), I must say I dislike group properties and system properties even more. They are very hard to discover, there is no IDE support for configuring them and most of them are not documented. And system properties make it impossible to have two instances running on one JVM with different settings.
If it is possible, let's add XML and java config.

Reply:
Not possible because there is no reference to config when SqlPredicate gets parsed.